pkg/services/accesscontrol/ossaccesscontrol/ossaccesscontrol_test.go

```diff
@@ -31,6 +32,30 @@ func setupTestEnv(t testing.TB) *OSSAccessControlService {
 	return &ac
 }
 
+func removeRoleHelper(role string) {
+	delete(accesscontrol.FixedRoles, role)
+
+	// Compute new grants removing any appearance of the role in the list
+	replaceGrants := map[string][]string{}
+	for builtInRole, grants := range accesscontrol.FixedRoleGrants {
+		n := len(grants)
+		for i := 0; i < n; i++ {
+			r := grants[i]
+			if r == role {
+				copy(grants[i:], grants[i+1:])
+				n--
+				grants = grants[:n]
+			}
+		}
```

Comment:
Golang strings are actually pointers, so this is actually doing a copy() on ints.  An assignment will have the same effect.

The non self modifying version is simpler and also doesn't segfault when removing the last element of the array.

Reply:
```suggestion
				func filterRoles(grants []string], removeRole string) []string {
            var out := make([]string, len(roles))
            for i, r := range grants {
                if r != role {
                    out = append(out, r)
                }
		    }
		return out
        }
    
```

Reply:
For extra details: https://commaok.xyz/post/intern-strings/

Reply:
I agree this version is simpler to read and I think more efficient as well. I'm going to make the change.
I'm not sure I understand the segfault statement though, I think it works alright: https://play.golang.org/p/mFMW8k-8wlZ